src/network/networkd-routing-policy-rule.c

```diff
@@ -0,0 +1,902 @@
+                        if (r < 0)
+                                continue;
+
+                        if (STR_IN_SET(a, "from", "to")) {
+
+                                r = in_addr_prefix_from_string(b, AF_INET, &buffer, &prefixlen);
+                                if (r < 0) {
+                                        r = in_addr_prefix_from_string(b, AF_INET6, &buffer, &prefixlen);
+                                        if (r < 0) {
+                                                log_error_errno(r, "RPDB rule prefix is invalid, ignoring assignment: %s", b);
+                                                continue;
+                                        }
+
+                                        rule->family = AF_INET6;
+                                } else
+                                        rule->family = AF_INET;
+
+                                if (streq(a, "to")) {
+                                        rule->to = buffer;
+                                        rule->to_prefixlen = prefixlen;
+                                } else {
+                                        rule->from = buffer;
+                                        rule->from_prefixlen = prefixlen;
+                                }
+                        } else if (streq(a, "tos")) {
+                                r = safe_atou8(b, &rule->tos);
+                                if (r < 0) {
+                                        log_error_errno(r, "Failed to parse RPDB rule tos, ignoring: %s", b);
+                                        continue;
+                                }
+                        } else if (streq(a, "table")) {
+                                r = safe_atou32(b, &rule->table);
+                                if (r < 0) {
+                                        log_error_errno(r, "Failed to parse RPDB rule table, ignoring: %s", b);
+                                        continue;
+                                }
+                        } else if (streq(a, "fwmark")) {
+                                char *q;
+
+                                q = strchr(b, '/');
+                                if (q)
+                                        *q++ = '\0';
+
+                                r = safe_atou32(b, &rule->fwmark);
+                                if (r < 0) {
+                                        log_error_errno(r, "Failed to parse RPDB rule firewall mark, ignoring: %s", b);
+                                        continue;
+                                }
+
+                                if (q) {
+                                        r = safe_atou32(q, &rule->fwmask);
+                                        if (r < 0) {
+                                                log_error_errno(r, "Failed to parse RPDB rule mask, ignoring: %s", p);
+                                                continue;
+                                        }
+                                }
```

Comment:
we probably should have a parse_fwmark_fwmask() function doing this, no? One that makes sure the ranges match, and the bits set too?